File: output/gpt_4o/avatar/algo_based_translation/python/java/atcoder_AGC010_B.java

import java.util.Scanner;

public class atcoder_AGC010_B {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int n = scanner.nextInt();
        scanner.nextLine(); // consume the leftover newline
        String[] input = scanner.nextLine().split(" ");
        int[] a = new int[n];

        for (int i = 0; i < n; i++) {
            a[i] = Integer.parseInt(input[i]);
        }

        int sumA = 0;
        for (int num : a) {
            sumA += num;
        }

        if (sumA % (n * (n + 1) / 2) != 0) {
            System.out.println("NO");
        } else {
            double k = sumA / (double) (n * (n + 1) / 2);

            int[] aExtended = new int[n + 1];
            System.arraycopy(a, 0, aExtended, 0, n);
            aExtended[n] = a[0];

            int sumMod = 0;
            int sumDiv = 0;

            for (int i = 0; i < n; i++) {
                int diff = (int) (k - aExtended[i] + aExtended[i + 1]);
                sumMod += diff % n;
                if (diff / n >= 0) {
                    sumDiv++;
                }
            }

            if (sumMod != 0 || sumDiv != n) {
                System.out.println("NO");
            } else {
                System.out.println("YES");
            }
        }
    }
}

File: dataset/avatar/Python/Code/atcoder_AGC010_B.py
n , a = int ( input ( ) ) , list ( map ( int , input ( ).split ( ) ) )
if sum ( a ) % ( n * ( n + 1 ) / 2 ) :
    print ( "NO" )
else :
    k = sum ( a ) / ( n * ( n + 1 ) / 2 )
    a += [ a [ 0 ] ]
    if sum ( [ ( k - i + j ) % n for i , j in zip ( a [ 1 : ] , a ) ] ) or sum ( [ ( k - i + j ) / n >= 0 for i , j in zip ( a [ 1 : ] , a ) ] ) != n :
        print ( "NO" )
    else :
        print ( "YES" )

